java/configuration/src/main/java/sleeper/configuration/properties/table/TableProperty.java

```diff
@@ -54,145 +57,188 @@ public interface TableProperty extends SleeperProperty {
             .defaultProperty(DEFAULT_SIZERATIO_COMPACTION_STRATEGY_MAX_CONCURRENT_JOBS_PER_PARTITION)
             .description("Used by the SizeRatioCompactionStrategy to control the maximum number of jobs that can be running " +
                     "concurrently per partition.")
+            .propertyGroup(TablePropertyGroup.COMPACTION)
             .build();
 
     // System defined
-    TableProperty SPLIT_POINTS_KEY = named("sleeper.table.splits.key").build();
-    TableProperty DATA_BUCKET = named("sleeper.table.data.bucket").build();
+    TableProperty SPLIT_POINTS_KEY = named("sleeper.table.splits.key")
+            .propertyGroup(TablePropertyGroup.SPLIT_POINTS).build();
+    TableProperty DATA_BUCKET = named("sleeper.table.data.bucket")
+            .propertyGroup(TablePropertyGroup.CONFIGURATION).build();
     // DynamoDBStateStore properties
-    TableProperty ACTIVE_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.active.table").build();
-    TableProperty READY_FOR_GC_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.gc.table").build();
-    TableProperty PARTITION_TABLENAME = named("sleeper.table.metadata.dynamo.partition.table").build();
+    TableProperty ACTIVE_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.active.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
+    TableProperty READY_FOR_GC_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.gc.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
+    TableProperty PARTITION_TABLENAME = named("sleeper.table.metadata.dynamo.partition.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
     TableProperty DYNAMO_STATE_STORE_POINT_IN_TIME_RECOVERY = named("sleeper.table.metadata.dynamo.pointintimerecovery")
             .defaultProperty(DEFAULT_DYNAMO_POINT_IN_TIME_RECOVERY_ENABLED)
+            .propertyGroup(TablePropertyGroup.METADATA)
             .build();
     TableProperty DYNAMODB_STRONGLY_CONSISTENT_READS = named("sleeper.table.metadata.dynamo.consistent.reads")
             .defaultProperty(DEFAULT_DYNAMO_STRONGLY_CONSISTENT_READS)
             .validationPredicate(s -> s.equalsIgnoreCase("true") || s.equalsIgnoreCase("false"))
+            .propertyGroup(TablePropertyGroup.METADATA)
             .build();
     // S3StateStore properties
-    TableProperty REVISION_TABLENAME = named("sleeper.table.metadata.s3.dynamo.revision.table").build();
+    TableProperty REVISION_TABLENAME = named("sleeper.table.metadata.s3.dynamo.revision.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
     TableProperty S3_STATE_STORE_DYNAMO_POINT_IN_TIME_RECOVERY = named("sleeper.table.metadata.s3.dynamo.pointintimerecovery")
             .defaultProperty(DEFAULT_DYNAMO_POINT_IN_TIME_RECOVERY_ENABLED)
+            .propertyGroup(TablePropertyGroup.METADATA)
             .build();
```

Comment:
I noticed that the default properties all have the same group, unlike in the table properties. Maybe make the group for the default instance properties explain that they're defaults for table properties?